Pathogens/Utility/location_utils/lat_lng_generation_report.py

```diff
@@ -0,0 +1,314 @@
+    details = '[City name: ' + city_name_as_string + ', State name: ' + state_name_as_string + ']'
+    suggestion = "[According to mapbox, the state this city belongs to is " + region_name_as_string + ". If there are no other errors and you agree with this state placement, consider moving the city to that state. If that state name sounds wrong, please feel free to omit the city name since it might be resulting in the pin being in the incorrect state. Please also check the other errors to see if they provide any more context here.]"
+
+    return prefix + " - " + info_statement + " - " + details + " - " + suggestion
+
+def generate_line_for_invalid_city_but_valid_state_check(
+    city_name: str | None,\
+    state_name: str | None,\
+    country_name: str,\
+    mapbox_response,\
+):
+    if(city_name is None):
+        return
+
+    if(mapbox_response is not None and (mapbox_response.text == city_name or mapbox_response.matching_text == city_name)):
+        return
+
+    mapbox_state_info_response = mapbox_api_client.make_mapbox_request(
+        city_name = None,
+        state_name = city_name,
+        country_name = country_name,
+        lat_lng_report_file = None,
+        mapbox_request_param_override = None
+    )
+    
+    if(mapbox_state_info_response is None):
+        return
+    
+    original_request_text_as_string = format_mapbox_text_and_matching_text_for_display(mapbox_response)
+    city_name_as_string = ('"' + city_name + '"') if city_name is not None else "N/A"
+    state_name_as_string = ('"' + state_name + '"') if state_name is not None else "N/A"
+    state_name_from_mapbox_as_string = format_mapbox_text_and_matching_text_for_display(mapbox_state_info_response)
+
+    prefix = generate_mapbox_api_request_log_prefix(
+        logLevel = MapboxApiRequestLogLevel.WARN,
+        city_name = city_name,
+        state_name = state_name,
+        country_name = country_name
+    )
+
+    info_statement = "[City listed is recognized as both a valid city and a valid state.]" if(mapbox_response is not None) \
+        else "[City listed is not a valid city, but is a valid state]"
+    details = '[City name given: ' + city_name_as_string + ', State name given: ' + state_name_as_string + ']'
+    suggestion = "[According to mapbox, the city you've given is the name of a valid state called " + state_name_from_mapbox_as_string + ". This might be resulting in the pin being in the wrong location because it might be using a city whose name is similar instead of using the state. The city name mapbox lists for this city is " + original_request_text_as_string + " so if that sounds wrong I would recommend moving the current city name to the state column in Airtable if that seems appropriate and makes sense given the other error messages.]"
+
+    return prefix + " - " + info_statement + " - " + details + " - " + suggestion
+
+def generate_line_for_invalid_city_but_valid_district_check(
```

Comment:
Context: Same problem as the state check above. Giving a district name as a city name only works if that city is also a district which is usually true but not always and when it's not true Mapbox just takes it's best guess and is almost always in the wrong part of the country.

I really feel like there should be a "District" column in Airtable so that we can get cases like these right but again I've gone super down the rabbit hole so sorry if my rambling doesn't make sense. I'll talk to the data team about it, I just think this report is valuable for illustrating my point.